src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-spacy.py

```diff
@@ -0,0 +1,20 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2020 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the Apache License 2.0
+#
+# The full license is available in LICENSE.APL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: Apache-2.0
+# ------------------------------------------------------------------
+
+"""
+Spacy contains hidden imports and data files which are needed to import it
+"""
+
+from PyInstaller.utils.hooks import collect_data_files, collect_submodules
+
+datas = collect_data_files("spacy")
+hiddenimports = collect_submodules("spacy")
+hiddenimports += ['srsly.msgpack.util', 'cymem.cymem', 'preshed.maps']
```

Comment:
These additional hiddenimports aren't coming from `spacy˙.

`srsly.msgpack.util` is required by `srsly.msgpack˙ (or more specifically, by `srsly.msgpack._packer` extension).

`preshed.maps˙ and `cymem.cymem` are coming from `thinc.backends.numpy_ops` extensions.

As such, I'd prefer to see these as separate hooks, so they can also be used outside of `spacy`.

Reply:
@rokm - for my own education, would you mind explaining how you determined where the hidden imports really came from? Happy to make these updates in the meantime!

Reply:
> @rokm - for my own education, would you mind explaining how you determined where the hidden imports really came from? Happy to make these updates in the meantime!

You can see the import chain in the traceback. E.g., for a simple program that imports `spacy`, you get this:

```
Traceback (most recent call last):
  File "program.py", line 1, in <module>
    import spacy
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "spacy/__init__.py", line 10, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "thinc/__init__.py", line 5, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "thinc/config.py", line 14, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "srsly/__init__.py", line 4, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "srsly/_msgpack_api.py", line 3, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 540, in exec_module
  File "srsly/msgpack/__init__.py", line 8, in <module>
  File "srsly/msgpack/_packer.pyx", line 7, in init srsly.msgpack._packer
ModuleNotFoundError: No module named 'srsly.msgpack.util'
```

So the import chain is: program -> `spacy` -> `thinc` -> `thinc.config` -> `srsly` -> `srsly._msgpack_api` -> `srsly.msgpack` -> `srsly.msgpack._packer` (which has .pyx extension, so it's an extension module, and PyInstaller can't pick up its imports automatically) -> `srsly.msgpack.util`

Reply:
Got it! Thanks for clarifying.